clippy_lints/src/functions/mod.rs

```diff
@@ -326,6 +327,25 @@ declare_clippy_lint! {
     "getter method returning the wrong field"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Detects lifetimes that are secretly `'static`.
+    /// ### Why is this bad?
+    /// One could think that the lifetime is dropped eventually, but it isn't, as it is really a `'static`
```

Comment:
More cents: Similar to above, language "lifetime is dropped": lifetimes are never dropped. Maybe: "The introduction of a lifetime parameter that can only resolve to `'static` adds unnecessary complexity."

Reply:
Suggestion applied.